connect/src/routes/request.ts

```diff
@@ -86,15 +86,21 @@ export class RouteTransferRequest<N extends Network> {
     params: {
       source: TokenId<FC>;
       destination: TokenId<TC>;
+      sourceDecimals?: number;
```

Comment:
We can pass in the decimals as an optimization (so we don't have to fetch if already known) and also when the token doesn't exist yet (e.g. unattested destination token) which would throw an error.